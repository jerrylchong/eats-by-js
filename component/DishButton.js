import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity, Dimensions} from 'react-native';

const DishButton = (props) => {

    const { title, description, price, onPress } = props;

    return (
        <TouchableOpacity style = {styles.container} onPress = {onPress}>
            <View style={styles.text}>
                <Text style={styles.name}>{title}</Text>
                <Text style={styles.cost}>${price}</Text>
            </View>
            <Text style = {styles.description}>{description}</Text>
        </TouchableOpacity>
    )
}

export default DishButton

const styles = StyleSheet.create({
    container: {
        paddingHorizontal:"3%",
        width: '100%',
        height: Dimensions.get('window').height * 0.10,
        alignSelf: 'center',
        justifyContent: 'center',
        borderBottomWidth: 0.5,
        borderColor: '#b3b3b3',
        backgroundColor: 'white'
    },
    text: {
        width: '100%',
        height: '30%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginBottom: '5%',
        marginTop: '2%'
    },
    nameReview: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    name: {
        color: 'black',
        fontSize: 20,
        fontFamily: 'Ubuntu-Bold'
    },
    cost: {
        color: 'black',
        fontSize: 20,
        fontFamily: 'Ubuntu'
    },
    description: {
        color: '#7e7e7e',
        fontSize: 11,
        alignSelf: 'flex-start',
        fontFamily: 'Ubuntu'
    },
})
