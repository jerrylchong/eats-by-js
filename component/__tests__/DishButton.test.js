import React from 'react';
import renderer from 'react-test-renderer';
import DishButton from "../DishButton";

test('renders correctly', () => {
    const tree = renderer.create(<DishButton/>).toJSON();
    expect(tree).toMatchSnapshot();
})