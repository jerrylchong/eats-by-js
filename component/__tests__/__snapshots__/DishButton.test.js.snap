// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderBottomWidth": 0.5,
      "borderColor": "#b3b3b3",
      "height": 133.4,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": "3%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": "30%",
        "justifyContent": "space-between",
        "marginBottom": "5%",
        "marginTop": "2%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Ubuntu-Bold",
          "fontSize": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Ubuntu",
          "fontSize": 20,
        }
      }
    >
      $
    </Text>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#7e7e7e",
        "fontFamily": "Ubuntu",
        "fontSize": 11,
      }
    }
  />
</View>
`;
